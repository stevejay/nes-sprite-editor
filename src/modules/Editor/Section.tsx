import React from "react";
import styles from "./Section.module.scss";

type Props = {
  children: React.ReactNode;
};

const Section: React.FunctionComponent<Props> = ({ children }) => (
  <section className={styles.section}>{children}</section>
);

export default Section;
