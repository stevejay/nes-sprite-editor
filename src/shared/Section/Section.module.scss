@import "styles/_base.scss";

.section {
  padding: $gutter / 2;
}

.section h2 {
  font-size: rem(20);
}

.section h3 {
  font-size: 1rem;
  font-weight: $font-weight-regular;
  padding-bottom: rem(4);
  border-bottom: 1px dashed $md-grey-500;
  text-transform: uppercase;
}
