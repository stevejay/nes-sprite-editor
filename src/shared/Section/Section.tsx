import React from "react";
import styles from "./Section.module.scss";
import classNames from "classnames";

type Props = {
  className?: string;
  children: React.ReactNode;
};

const Section = ({ className, children }: Props) => {
  const sectionClassName = classNames(styles.section, className);
  return <section className={sectionClassName}>{children}</section>;
};

export default Section;
