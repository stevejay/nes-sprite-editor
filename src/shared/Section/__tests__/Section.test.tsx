import React from "react";
import { render, cleanup } from "react-testing-library";
import Section from "../Section";

afterEach(cleanup);

test("displays correctly", async () => {
  const { container } = render(<Section>content</Section>);
  expect(container).toMatchSnapshot();
});
