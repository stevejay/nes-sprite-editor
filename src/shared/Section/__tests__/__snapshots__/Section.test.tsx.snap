// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly 1`] = `
<div>
  <section
    class="section"
  >
    content
  </section>
</div>
`;
